wdl-analysis/src/graph.rs

```diff
@@ -629,3 +629,40 @@ impl DocumentGraph {
             .retain(|(from, to)| !collected.contains(from) && !collected.contains(to));
     }
 }
+
+/// An extension trait for [`DocumentGraph`].
+pub trait DocumentGraphExt {
+    /// Gets all nodes that have a dependency on the given node.
+    fn dependents(
+        &self,
+        index: petgraph::graph::NodeIndex,
+    ) -> Box<dyn Iterator<Item = petgraph::graph::NodeIndex> + '_>;
+}
+
+impl DocumentGraphExt for DocumentGraph {
+    fn dependents(
```

Comment:
Above, we already have `DocumentGraph::dependencies()` that gets the immediate incoming neighbors of a node. Calling this `dependents()` by analogy implies that it does not do a full traversal for all the dependents. I recommend we call this something like `transitive_dependents()` or `all_dependents()` instead to differentiate.